api/openapi/specs/core_service.yaml

```diff
@@ -0,0 +1,331 @@
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "./api_types.yaml#/components/schemas/ServiceIdentifiers"
+
+  /enclaves/{enclave_identifier}/services:
+    get:
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - in: query
+          name: services
+          schema:
+            type: array
+            items:
+              type: string
+          description: Select services to get information
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties:
+                  $ref: "./api_types.yaml#/components/schemas/ServiceInfo"
+
+  /enclaves/{enclave_identifier}/services/{service_identifier}/command:
+    post:
+      description: Executes the given command inside a running container
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - $ref: './api_types.yaml#/components/parameters/service_identifier'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "./api_types.yaml#/components/schemas/ExecCommand"
+        description: Exec Command
+        required: true
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                $ref: "./api_types.yaml#/components/schemas/ExecCommandResult"
+
+  /enclaves/{enclave_identifier}/services/{service_identifier}/endpoints/{port_number}/availability:
+    get:
+      description: Block until the given HTTP endpoint returns available, calling it through a HTTP request
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - $ref: './api_types.yaml#/components/parameters/service_identifier'
+        - $ref: './api_types.yaml#/components/parameters/port_number'
+        - $ref: './api_types.yaml#/components/parameters/http_method'
+        - $ref: './api_types.yaml#/components/parameters/path'
+        - $ref: './api_types.yaml#/components/parameters/initial_delay_milliseconds'
+        - $ref: './api_types.yaml#/components/parameters/retries'
+        - $ref: './api_types.yaml#/components/parameters/retries_delay_milliseconds'
+        - $ref: './api_types.yaml#/components/parameters/expected_response'
+        - $ref: './api_types.yaml#/components/parameters/request_body'
```

Comment:
my only concern here is that GET request has a char limits depending on browsers I think, and the request body could be long, but probably is ok for now and if we see users having this problem we can move it to a POST request maybe

Reply:
Fair point! I think the limit is 2048 but most of these fields are optional and we don't expect them to any large since those are for status checking endpoints. From the RESTful perspective, and GET makes more sense since they have no side effect in the backend. Let's leave as a GET for now, and we can reconsider it if we actually start runinig into issues.